app/controllers/concerns/my_facilities_filtering.rb

```diff
@@ -5,38 +5,49 @@ module MyFacilitiesFiltering
   NEW_FACILITY_THRESHOLD = 3.months.ago
 
   included do
-    before_action :populate_facility_sizes, :populate_zones,
-      :set_selected_sizes, :set_selected_zones, :set_only_new_facilities
-
-    def filter_facilities(scope_namespace = [])
-      facilities = current_admin.accessible_facilities(:view_reports)
-
-      filtered_facilities = facilities_by_size(facilities)
+    before_action :populate_facility_groups
+    before_action :set_selected_facility_group
+    before_action :populate_facilities
+    before_action :populate_facility_sizes
+    before_action :populate_zones
+    before_action :set_selected_facility_sizes
+    before_action :set_selected_zones
+
+    def filter_facilities
+      filtered_facilities = facilities_by_size(@facilities)
       facilities_by_zone(filtered_facilities)
     end
 
     private
 
+    def populate_facility_groups
+      @facility_groups = current_admin.accessible_facility_groups(:view_reports).order(:name)
+    end
+
     def populate_facility_sizes
       @facility_sizes = Facility.facility_sizes.keys.reverse
     end
 
+    def populate_facilities
+      @facilities = current_admin.accessible_facilities(:view_reports).where(facility_group: @selected_facility_group)
+    end
+
     def populate_zones
-      @zones = current_admin.accessible_facilities(:view_reports).pluck(:zone).uniq.compact.sort
+      @zones = @facilities.pluck(:zone).uniq.compact.sort
     end
 
-    def set_selected_sizes
+    def set_selected_facility_group
+      @selected_facility_group = params[:facility_group] ? @facility_groups.find_by(slug: params[:facility_group]) : @facility_groups.first
```

Comment:
With respect to ordering mindfully - is there a better default we can choose than `facility_groups.first`?

Reply:
That's a good question @harimohanraj89 — the facility groups are sorted by alphabetical order and I think the first district in that list makes most sense since the list only include facility groups the admin has access to. Do you have an idea for a better default?

Reply:
Nope, I don't 😅  I also kinda missed the fact that these will be alphabetically sorted. Sounds good to me!